include/random.h

```diff
@@ -73,14 +80,34 @@ enum RandomTag
 
 #define RandomPercentage(tag, t) \
     ({ \
-        const u8 weights[] = { 100 - t, t }; \
-        RandomWeightedArray(tag, 100, ARRAY_COUNT(weights), weights); \
+        u32 r; \
+        if (t <= 0) \
+        { \
+            r = FALSE; \
+        } \
+        else if (t >= 100) \
+        { \
+            r = TRUE; \
+        } \
```

Comment:
This fixes abilities like Compound Eyes which produce a "percentage" that is out of bounds.